tcms/testplans/tests/test_new_plan.py

```diff
@@ -109,3 +111,49 @@ def test_post_with_no_perm_redirects_to_login(self):
         response = self.client.post(self.location, self.request, follow=True)
 
         self.assertRedirects(response, reverse('tcms-login') + '?next=' + self.location)
+
+    def test_with_invalid_product_shows_error(self):
+        new_data = self.request.copy()
+        new_data['product'] = -1
+        del new_data['email_settings-0-auto_to_plan_author']
+
+        response = self.client.post(self.location, data=new_data, follow=True)
+
+        self.assertContains(response, _('Create new TestPlan'))
+        self.assertContains(
+            response,
+            _('Select a valid choice. That choice is not one of the available choices.'))
+        self.assertContains(
+            response,
+            '<input class="bootstrap-switch" name="email_settings-0-auto_to_plan_author" '
+            'type="checkbox"', html=False)
+
+    def test_with_invalid_type_shows_error(self):
+        new_data = self.request.copy()
+        new_data['type'] = -1
+        del new_data['email_settings-0-auto_to_case_owner']
+
+        response = self.client.post(self.location, data=new_data, follow=True)
+
+        self.assertContains(response, _('Create new TestPlan'))
+        self.assertContains(
+            response,
+            _('Select a valid choice. That choice is not one of the available choices.'))
+        self.assertContains(
+            response,
+            '<input class="bootstrap-switch" name="email_settings-0-auto_to_case_owner" '
+            'type="checkbox"', html=False)
+
+    def test_with_invalid_version_shows_error(self):
+        new_data = self.request.copy()
+        # Note: version not assigned to the current Product, that's why
+        # it is invalid !
+        version = VersionFactory()
+        new_data['product_version'] = version.pk
+
+        response = self.client.post(self.location, data=new_data, follow=True)
+
+        self.assertContains(
+            response,
+            _('Select a valid choice. That choice is not one of the available choices.'))
```

Comment:
All of these 3 scenarios are exactly the same - trigger a form error which displays an error. They just do it via different fields.

However this part is handled internally by Django so we can argue that we don't really need to test that b/c Django already does.   At most only 1 of these scenarios should stay if you are keen on keeping them.